code/game/machinery/vending/vendor_types/food.dm

```diff
@@ -9,20 +9,12 @@
 	unslashable = FALSE
 	wrenchable = TRUE
 
-	listed_products = list( // see also /datum/decorator/christmas/food
-			list("PREPARED MEALS", -1, null, null),
-			list("USCM Prepared Meal (Chicken)", 15, /obj/item/reagent_container/food/snacks/mre_pack/meal5, VENDOR_ITEM_REGULAR),
-			list("USCM Prepared Meal (Cornbread)", 15, /obj/item/reagent_container/food/snacks/mre_pack/meal1, VENDOR_ITEM_REGULAR),
-			list("USCM Prepared Meal (Pasta)", 15, /obj/item/reagent_container/food/snacks/mre_pack/meal3, VENDOR_ITEM_REGULAR),
-			list("USCM Prepared Meal (Pizza)", 15, /obj/item/reagent_container/food/snacks/mre_pack/meal4, VENDOR_ITEM_REGULAR),
-			list("USCM Prepared Meal (Pork)", 15, /obj/item/reagent_container/food/snacks/mre_pack/meal2, VENDOR_ITEM_REGULAR),
-			list("USCM Prepared Meal (Tofu)", 15, /obj/item/reagent_container/food/snacks/mre_pack/meal6, VENDOR_ITEM_REGULAR),
-			list("USCM Protein Bar", 50, /obj/item/reagent_container/food/snacks/protein_pack, VENDOR_ITEM_REGULAR),
-
-			list("FLASKS", -1, null, null),
-			list("Metal Flask", 10, /obj/item/reagent_container/food/drinks/flask, VENDOR_ITEM_REGULAR),
-			list("USCM Flask", 5, /obj/item/reagent_container/food/drinks/flask/marine, VENDOR_ITEM_REGULAR)
-			)
+/obj/structure/machinery/cm_vending/sorted/marine_food/get_listed_products(mob/user)
+	. = list()
+	. += GLOB.cm_vending_marine_food
+	//. += GLOB.cm_vending_marine_food_christmas
+	. += GLOB.cm_vending_marine_food_flasks
```

Comment:
mind this creates a new list so should be avoided for bigger vendors

should integrate with the xmas food decorator somehow if possible

Reply:
I've reworked this so the food is added to the populate_product_list proc - this is only ran on init, so the decorator will override the listed_products.